cloudinit/config/schemas/schema-cloud-config-v1.json

```diff
@@ -2214,6 +2214,110 @@
           "additionalProperties": false
         }
       }
+    },
+    "reporting_config": {
+      "type": "object",
+      "properties": {
+        "reporting": {
+          "type": "object",
+          "additionalProperties": false,
+          "patternProperties": {
+            "^.+$": {
+              "label": "<arbitrary_name>",
+              "type": "object",
+              "oneOf": [
+                {
+                  "additionalProperties": false,
+                  "properties": {
+                    "required": ["type"],
+                    "type": {
+                      "type": "string",
+                      "enum": ["log"]
+                    },
+                    "level": {
+                      "type": "string",
+                      "enum": ["DEBUG", "INFO", "WARN", "ERROR", "FATAL"],
+                      "default": "DEBUG"
+                    }
+                  }
+                },
+                {
+                  "additionalProperties": false,
+                  "properties": {
+                    "required": ["type"],
```

Comment:
I think the "type" property is implicitly checked here without extra effort, but gives a slightly less valuable error message. The reason is that there is only one property key and absence of any properties in a config will be evaluated as nonetype/null. I expect will that current validation causes a warning like `is not of type 'object'`, rather than the marginally better user message: `not valid`.

I'm only adding this detail because this may require updating some unit tests match statements as well as the change here.